src/NuGet.Core/NuGet.Packaging/TopologicalSortUtility.cs

```diff
@@ -10,24 +10,21 @@ namespace NuGet.Packaging
 {
     public static class TopologicalSortUtility
     {
+        private static readonly PackageInfoComparer DefaultComparer = new PackageInfoComparer();
         /// <summary>
         /// Order dependencies by children first.
         /// </summary>
         public static IReadOnlyList<PackageDependencyInfo> SortPackagesByDependencyOrder(
             IEnumerable<PackageDependencyInfo> packages)
         {
-            var sorted = new List<PackageDependencyInfo>();
-            var toSort = packages.Distinct().ToList();
+            var toSort = packages.Distinct().Select(package => new PackageInfo(package)).ToArray();
```

Comment:
Measurements are earlier comment above

TopologicalSortCurrent == current code
TopologicalSortReduced == this PR

For a single entry, so single loop iteration allocates less:
```
                 Method | Deps |     Gen 0 |   Gen 1 |    Allocated |
----------------------- |----- |----------:|--------:|-------------:|
 TopologicalSortCurrent |    1 |    0.3719 | 62.5000 |      1,176 B |
 TopologicalSortReduced |    1 |    0.2642 |       - |        832 B |
```
Increasing the dependencies, you can see the allocations are definitely on different growth curves
```
                 Method | Deps |     Gen 0 |   Gen 1 |    Allocated |
----------------------- |----- |----------:|--------:|-------------:|    
 TopologicalSortCurrent |  512 | 4062.5000 | 62.5000 | 12,874,392 B |
 TopologicalSortReduced |  512 |         - |       - |    181,016 B |
```